include/quicr/detail/ctrl_message_types.h

```diff
@@ -84,61 +87,96 @@ namespace quicr::messages {
             }
             return size;
         }
+
+        /**
+         * Equality comparison operator for KeyValuePair.
+         * @param other The KeyValuePair to compare with.
+         * @return True if both KeyValuePair objects are equal, false otherwise.
+         */
+        bool operator==(const KeyValuePair<T>& other) const
+        {
+            if (type != other.type) {
+                return false;
+            }
+
+            if (static_cast<std::uint64_t>(type) % 2 != 0) {
+                // Odd types are byte equality.
+                return value == other.value;
+            }
+
+            // Even types are numeric equality.
+            if (value.size() > sizeof(std::uint64_t) || other.value.size() > sizeof(std::uint64_t)) {
+                throw std::invalid_argument("Even KVPs must be <= 8 bytes");
+            }
+
+            // Compare numeric values.
+            const auto smaller = std::min(value.size(), other.value.size());
+            if (memcmp(value.data(), other.value.data(), smaller) != 0) {
+                return false;
+            }
+
+            // Are there left over bytes to check?
+            const auto larger = std::max(value.size(), other.value.size());
+            if (larger == smaller) {
+                return true;
+            }
+
+            // Any remaining bytes could be 0, but nothing else.
+            const auto longer = (value.size() > other.value.size()) ? value : other.value;
```

Comment:
Since these are not trivial types (vectors), shouldn't this be 
```suggestion
            const auto& longer = (value.size() > other.value.size()) ? value : other.value;
```

though these are small vectors, so maybe it doesn't matter.